pkgs/by-name/cu/cups-idprt/package.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, cups
+, unzip
+}:
+
+stdenv.mkDerivation {
+  pname = "cups-idprt";
+  version = "1.4.2";
+
+  src = fetchurl {
+    url = "https://www.idprt.com/prt_v2/files/down_file/id/131/fid/431.html"; # NOTE: This is NOT an HTML page, but a ZIP file
+    sha256 = "sha256-xY5E1RUNQwK0BO+0JMT7oHCAvSe79t7UU65IOzRQs/Q=";
+  };
+
+  buildInputs = [
+    cups
+  ];
+  nativeBuildInputs = [
+    autoPatchelfHook
+    unzip
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    unzip $src -d .
+    runHook postUnpack
+  '';
+
+  preBuild = ''
+    addAutoPatchelfSearchPath ${cups}/lib
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/cups/model
+    mkdir -p $out/lib/cups/filter
+    cp -r idprt_tspl_printer_linux_driver_v1.4.2/filter/${
+      if stdenv.hostPlatform.system == "x86_64-linux" then "x64"
+      else if stdenv.hostPlatform.system == "x86-linux" then "x86"
+      else throw "Unsupported host platform. Must be either 64-bit or 32-bit x86"
+    }/. $out/lib/cups/filter
```

Comment:
not sure what maintainers would think about this, but personally, i would extract to a variable for readability

Reply:
I think I found a better way.